src/core/qgspointlocator.h

```diff
@@ -446,9 +446,17 @@ class CORE_EXPORT QgsPointLocator : public QObject
     /**
      * find out if the \a point is in any polygons
      * This method is either blocking or non blocking according to \a relaxed parameter passed
+     * \deprecated QGIS 3.34 - use the overloaded pointInPolygon accepting a QgsPointLocator::MatchFilter parameter (pointInPolygonV2 in python)
      */
-    //!
-    MatchList pointInPolygon( const QgsPointXY &point, bool relaxed = false );
+    Q_DECL_DEPRECATED MatchList pointInPolygon( const QgsPointXY &point, bool relaxed = false ) SIP_DEPRECATED;
+
+    /**
+     * Find out if the \a point is in any polygons
+     * This method is either blocking or non blocking according to \a relaxed parameter passed
+     * Optional filter may discard unwanted matches.
+     * \since QGIS 3.34
+     */
+    MatchList pointInPolygon( const QgsPointXY &point, QgsPointLocator::MatchFilter *filter = nullptr, bool relaxed = false ) SIP_PYNAME( pointInPolygonV2 );
```

Comment:
Could we avoid the deprecation by just adding QgsPointLocator::MatchFilter *filter = nullptr as the new last argument to the existing method?

Reply:
I guess we could, but all other signatures follow this order in their optional parameters, it's a little disturbing to have this one the other way round!

Reply:
In that case I'd just add a "Todo qgis 4 - reorder to match .." note. It's ugly either way! 🤪

Reply:
+1 I would prefer ugly (filter as the last argument) than a potential API break / source of crashes (filter as the second argument) :slightly_smiling_face: 

Alternatively we could have a new function pointInPolygonWithFilter to side step this for the time being.

Reply:
Practically in python that is a new function as it's `SIP_PYNAME( pointInPolygonV2 )`.
I'll go with putting the filter as a last parameter and adding a notice on the docstring. I doubt those methods/parameters are used much in plugins anyway!